site/cypress/e2e/sidebar.spec.js

```diff
@@ -0,0 +1,83 @@
+import { getCurrentSemester } from '../support/utils.js';
+
+const currentSemester = `${getCurrentSemester()}`;
+
+Cypress.Commands.add('sidebarContains', (title, extension) => {
+    cy.get('aside ul li').contains(title).should('have.attr', 'href').and('contain', extension);
+    if (!title.includes('Logout')) {
+        cy.get('aside ul li').contains(title).click({ force: true });
+        if (title.includes('Autograding')) {
+            cy.get('#main > .content > h1').should('contain', 'Job Statistics');
+        }
+        else if (title.includes('Discussion Forum')) {
+            cy.get('#main > .content').should('contain', 'Create Thread');
+        }
+        else if (title.includes('Collapse Sidebar')) {
+            return;
+        }
+        else {
+            cy.get('#main > .content').should('contain', title);
+        }
+    }
```

Comment:
Complicated logic like this is prone to failure.  What prevents something from changing and accidentally cutting off this branch, leaving your tests un-run?

I think it would be good to cut back on the branches here (and elsewhere in the custom commands you've created), but I can't think of any specific suggestions as for how to make it better.  If you aren't able to find something cleaner without a significant amount of effort, this is probably good enough.